controller/pkg/extensions/merge.go

```diff
@@ -0,0 +1,23 @@
+package extensions
+
+import (
+	"github.com/samber/lo"
+
+	commonv1alpha1 "github.com/kong/kubernetes-configuration/api/common/v1alpha1"
+)
+
+// MergeExtensions merges the default extensions with the extensions provided by the user.
+// The provided extensions take precedence over the default extensions: in case
+// the user provides an extension that is also present in the default extensions,
+// the user's extension will be used.
+func MergeExtensions(defaultExtensions, extensions []commonv1alpha1.ExtensionRef) []commonv1alpha1.ExtensionRef {
```

Comment:
Could we add a UT for this?

Reply:
6966c2a4a7b44596a85d9daae3f2fed31dbaca0c